internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/DaVinciBlobFinder.java

```diff
@@ -22,32 +27,46 @@
 /**
  * DvcBlobFinder discovers live DaVinci peer nodes to facilitate blob transfers necessary for bootstrapping the database
  */
-public class DvcBlobFinder implements BlobFinder {
-  private static final Logger LOGGER = LogManager.getLogger(DvcBlobFinder.class);
+public class DaVinciBlobFinder implements BlobFinder {
+  private static final Logger LOGGER = LogManager.getLogger(DaVinciBlobFinder.class);
   private static final String TYPE_BLOB_DISCOVERY = "blob_discovery";
   private static final String ERROR_DISCOVERY_MESSAGE =
       "Error finding DVC peers for blob transfer in store: %s, version: %d, partition: %d";
-  private final TransportClient transportClient;
+  private final AvroGenericStoreClientImpl storeClient;
 
-  public DvcBlobFinder(TransportClient transportClient) {
-    this.transportClient = transportClient;
+  public DaVinciBlobFinder(AvroGenericStoreClientImpl storeClient) {
+    this.storeClient = storeClient;
   }
 
   @Override
   public BlobPeersDiscoveryResponse discoverBlobPeers(String storeName, int version, int partition) {
```

Comment:
Can you add another logic in the caller of this discoverBlobPeers and iterate through all the available peers? And please print out the peer for better future debugging purpose. For now I think we can iterate all of them before real prod onboarding, but in the future stage, we should set a limit of how many peers we can try before we give up.

Reply:
No I mean in the caller of this method, for now after discovery, the current logic only try for the first peer it sees, however, it could be a dead instance or some other issues (for example in the future certain node hit throttling limit). Can you iterate through and try to fetch from them, once you found one working, you can quit the loop.

Reply:
And let's not log all the peers in the list for now. Only log the peer hostname you try in the loop.

Reply:
please double check if the logic i implemented would work, it passes the local tests